### DISCUSSION TITLE
Shiny - Push notification newbie question

### DISCUSSION THREAD

**Discussion by gabsamples6** [shinyorg/shiny#1329] [Q&A]
Hi 
I am giving shiny a go after been burnt in the past with Appcenter and onesignal support for maui/xamarin is quite poor.

I have read the documentation but I still have some doubts that I want to put closure to.

1. google-services.json  still required - correct ?  I included it.
2. Push Notifications foreground its not supported  ( could not find how) but quite few examples googling using firebase directly)
3. OnEntry  never fires when app in either foreground or background..  I thought it should when a user taps on a notification
4. OnReceived only fires when app in foreground but not in background.
5. I would be interested in using Native and do not add any other dependencies but not found a quicky noddy sample that explains how you make it run.

I build a noddy example using the shiny template and then used firebase console to send notifications using firebase.

Am I missing the obvious with above..

Lastly - I could not find any sample that I could just run and replace with my google-services.json etc..  (I would not mind uploading one somewhere) once I have it working in both android and ios... 

In case I have missed very clear documentation that explains the above - please could you point me to a link?

Many thanks

**Comment by aritchie** [comment]
1. google-services.json is still fully support.  Look at the arguments registration process.
2. It is supported, but you have to do the mechanics to get the message to your UI.
3. OnEntry works fine.  Please read shinylib.net FAQ's and review the sample.  
4. It fires in both as long as you meet the criteria.  Also in in the FAQ.
5. I don't understand the question

shinylib.net is listed on the readme.md of this site.  Beyond that, it is open code.

**Comment by gabsamples6** [comment]
@aritchie  thanks for your time and reply.

**Entry not firing** 
Added click_action - see below  but still entry does not fire.. and I am using your template so everything else is already set.
![image](https://github.com/shinyorg/shiny/assets/109357537/a1e58eec-5f65-4ffd-b943-60c144e26fed)

**Postman test**
Used postman to send a push notification  - set up headers accordingly... 
Message is received by the app - OnReceived fires - OnEntry does not fire - pop up is not displaying  if Data is added as per documentation 
but then how do you  send the "click_action"
{
 "to" : "deviceToken",
  "notification" : {
     "title": "This is a title",
     "message" : "This is a message"
     },
 "data" : {  
      "click_action": "SHINY_PUSH_NOTIFICATION_CLICK"
     }
}

**Foreground Service:**
Happy to add the mechanics - any clues? is there a Shiny class I need to call  

I totally appreciate that this is free, but if there is an example somewhere or a postman test or anything that would be brilliant..

Any suggestions are welcome

**Reply by aritchie** [reply]
https://shinylib.net/client/push/#android-1 

Please read the docs.  There is a link to the android docs where to put click_action

**Comment by gabsamples6** [comment]
@aritchie  I have tried all possible scenarios of sticking click_action in various places.. but its not hitting the entry.. 
Would be possible for you to post a snippet of that.. I might be missing the obvious...   
Using your template I assume that  you included all permissions required in the android manifests for notification to work.. I could not see "messaging._Event" but i suppose is done in code.. 

I appreciate this is free and I am not asking a step by step guide - but I just want to hit the ground running with the basic stuff..

**Comment by gabsamples6** [comment]
@aritchie  Hi managed to get the entry working - you sample that is installed with template is incomplete and even with the correct payload the  entry will never be called.  

The way this is structured that you can swap providers etc.. is excellent , however documentation and lack of decent simple working sample makes  the onboarding process hard. We could consider sponsoring because is only fair in my view but its tricky if all your answers are "read documentation" etc..  or dont want to provide a sample payload either.

Happy to contribute in way of documentation and make it easier for people like me to hit the ground running straight away  but I am not sure you actually want that or If I provide something you simple say "its wrong" but dont say what is wrong..

I hope you dont misunderstand what I am trying to say as I am well aware of the line between "Free OSS" and paid support.

**Reply by aritchie** [reply]
1. What's incomplete?  Have you filed it as a bug?  What version of the template do you have?  https://github.com/shinyorg/templates

2. You haven't filed any reproducible sample with an actual issue.  You've asked if you could be missing something... how am I supposed  to know?  If something is wrong with the documentation... what is it?  You didn't say.  I don't owe people docs... I'm doing the best I can with the time I have.  I KNOW how the thing works.  My base is covered.  Again, it's open source.  It is meant to help people get places faster, not cover a lack of knowledge on a topic.  

3. I didn't say anything was wrong.  I said read the docs because I cover a lot of the points you've been asking.  I linked the docs to android because I also don't always know off the top of my head what the answer may be for platform specific stuff.  I don't have to post full raw examples.  I give enough away for free.

You submitted a ton of questions which I answered and pointed you towards places to look.  I don't have an example of google-services.json... correct, but again - its not a lot of code to look through.  Do I have to do a specific set of examples that fit your expectations (answer: NO...I don't owe anybody anything).

So... don't come here with expectations or presume to tell me what I'm "doing wrong".  This isn't tech support and you've really expected me to answer without showing me anything.  I'm happy to answer questions about how something works, point you towards where to look, or give some brief tutorials on certain topics.

**Reply by gabsamples6** [reply]
you are really twisting all this round! No one is expecting support or anything but before sponsoring a project which is only fair - you have a trial  and see how it goes...

1.  I have the latest template
2.  I have scanned all your closed answers and you always use the same line "read the docs" but the docs are incomplete or its just a refernce kind of docs.. 
3. I submitted questions because the sample was not working  and still is not working unless you tweak it.

there are no expectations and I am very respectful of the hours people like you put and for free ...  but a bare minimum to get going is required.. and noone is telling you  that "you are doing wrong" is the other way round, which is fine.. in many closed issues or discussions you are always dragging in given an answer.  

Look you owe nothing - absolutely correct..  but please dont twist it around... because I have been very polite and still i appreciate your effort and I am not demanding anything -

**Reply by aritchie** [reply]
You aren't being polite.  You are wasting my time and debating an answer you aren't happy with.  I've given you answers or pointers in every response and I've been polite... short maybe... but you're repeatedly tagging me which is specifically requested you do not do in the forum guidelines: https://github.com/orgs/shinyorg/discussions/600 .  

You didn't answer ANY of my questions.  What's this "tweak" that I'm missing, so I can update the templates?  Where's the code issue with what's missing?  You keep saying the "docs are incomplete"?  What's missing... provide value or its just words and thus, a waste of time!

**Comment by gabsamples6** [comment]
Again you are twisting again. I honestly I was expecting answers from the community but you answered and of course I replied to you .its called manners not tagging . Anyway I won’t bother posting anymore. I am moving on . Thanks

**Comment by aritchie** [comment]
This user is banned for being annoying.  

* If anyone wants to tell me what more is needed in the docs, I'll see about getting it added.  I'm not documenting everything for both client (both platforms x multiple providers) & server because I simply don't have the time.  I provide links to the server docs!

* If anyone has a template bug where I'm missing something, let me know.  I will fix.  The template supports A LOT of features, so I miss things.  https://github.com/shinyorg/templates

* Provide some constructive feedback back other than "this is missing" or "this needs a tweak".  Provide said tweak and/or what is missing!  I'll fix it.  I'm trying, but mistakes happen.  

Lastly, don't offer to sponsor or contribute based on how much support I provide.  I'm not chasing sponsors, contributors, or a user base.  I just give this code away in hopes it helps others while helping myself.  If you expect more than the code, you've come to the wrong place.
